src/main/java/org/testng/IDataProviderListener.java

```diff
@@ -28,4 +28,17 @@ default void afterDataProviderExecution(
       IDataProviderMethod dataProviderMethod, ITestNGMethod method, ITestContext iTestContext) {
     // not implemented
   }
+
+  /**
+   * This method gets invoked when the data provider encounters an exception
+   *
+   * @param method - The {@link ITestNGMethod} method that received the data. A reference to the
+   *               corresponding data provider can be obtained via {@link ITestNGMethod#getDataProviderMethod()}
+   * @param ctx    - The current test context
+   * @param t      - The {@link RuntimeException} that embeds the actual exception. Use {@link
+   *               RuntimeException#getCause()} to get to the actual exception.
+   */
+  default void onDataProviderFailure(ITestNGMethod method, ITestContext ctx, RuntimeException t) {
```

Comment:
Maybe use `Throwable` instead of `RuntimeException `

Reply:
@juherr - I stuck to `RuntimeException` because that is what is being thrown as the wrapper and the actual problem gets embedded into it.